### PR TITLE
[Refactor](multi catalog)Split ExternalFileScanNode into FileQueryScanNode and FileLoadScanNode.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Split ExternalFileScanNode into FileQueryScanNode and FileLoadScanNode.
Remove some useless code in FileLoadScanNode.
Remove unused config item: enable_vectorized_load and enable_new_load_scan_node

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalScanNode.java [hunk 1]
```diff
@@ -33,10 +36,20 @@
...
-public class ExternalScanNode extends ScanNode {
+public abstract class ExternalScanNode extends ScanNode {
+    private static final Logger LOG = LogManager.getLogger(ExternalScanNode.class);
 
     // set to false means this scan node does not need to check column priv.
-    private boolean needCheckColumnPriv;
+    protected boolean needCheckColumnPriv;
+
+    // For explain
+    protected long inputSplitsNum = 0;
```
Comment (xxiao2018) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalScanNode.java:46]: I think this 4 fields should be in `FileScanNode`?

**Review by morningman — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalScanNode.java (hunk 1 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalScanNode.java:46]: Will fix this in next pr

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalScanNode.java [hunk 2]
```diff
@@ -21,9 +21,12 @@
 import org.apache.doris.planner.PlanNodeId;
 import org.apache.doris.planner.ScanNode;
 import org.apache.doris.statistics.StatisticalType;
-import org.apache.doris.thrift.TPlanNode;
 import org.apache.doris.thrift.TScanRangeLocations;
 
+import com.google.common.collect.Lists;
```
[the remark this answers is not in the document]
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalScanNode.java:26]: I didn't see a problem here and Check style didn't report any problem.

**Pull request merged**